http-support/src/site/markdown/index.md.vm

```diff
@@ -0,0 +1,340 @@
+# The HTTP support library
+
+During an ${docNameCap} execution it is typically necessary to communicate with 
+one or multiple HTTP servers. Examples include the download of artifacts from a
+repository or the interaction with an SW360 instance. Custom workflow step
+implementations may also contact specific servers to obtain additional 
+information about the software components used by a project.
+
+While doing HTTP requests is not rocket science, it is nevertheless 
+inefficient and error prone if every component that requires this
+communication comes with its own implementation. There are a number of aspects
+to be taken into account:
+
+* HTTP clients are often complex objects making use of expensive resources like
+  thread pools and connection pools. Having multiple clients in a single 
+  ${docNameCap} run is therefore a waste of resources, especially as HTTP clients are
+  typically thread-safe and can be shared between multiple components.
+* Configuring an HTTP client is not trivial; basic settings from the ${docNameCap}
+  configuration, such as Proxy settings or timeouts, should be taken into
+  account to make sure that all HTTP connections behave in a uniform way.
+* Resource and error handling when sending HTTP requests can become tricky. It
+  must be ensured that all resources are correctly released even in case of
+  failure responses and exceptions thrown during processing.
+* If many requests are to be sent - e.g. when downloading a number of 
+  artifacts -, performance can be improved by making use of threading and
+  parallelism. This complicates the programming model significantly though.
+ 
+To deal with these issues, ${docNameCap} provides a simple to use HTTP client
+library that can be used by all components taking part in an ${docNameCap} execution.
+The library manages a central, fully configured HTTP client that offers the
+following features:
+
+* HTTP requests can be defined in a declarative way making use of a domain
+  specific language (DSL).
+* The API is lambda-friendly; for instance the processing of a response can be
+  done by a lambda expression.
+* Resource management is handled by the library; client code does not have to
+  close responses or entities, or use means like try-with-resources.
+* The programming model is reactive using _CompletableFutures_ to allow for
+  the efficient processing of many requests. There is, however, an easy means
+  to support blocking calls as well.
+* JSON processing is directly supported, as this is a standard protocol used by
+  many APIs.
+* Support for multipart requests that are used for instance for file upload
+  operations.
+
+The library is implemented by the _http-support_ sub module of the ${docNameCap}
+project. It has comprehensive Javadoc which can be consulted for detail
+questions. This document provides a short user's guide.
+
+#[[##]]# Obtaining the HTTP client
+
+The HTTP client can be easily obtained from the _AntennaContext_, which is
```

Comment:
THis is the only reference to Antenna. Since the documentation is in the http library and the library is without any dependency to Antenna, I would prefer to not mention this here. This is actually something that should be documented in the AntennaContext docu.

Reply:
Done. The reference to AntennaContext has been removed from the documentation of the Http support library. A paragraph has been added to the Antenna developer guide that mentions that some shared service objects can be queried from the Antenna context.